src/NuGet.Clients/NuGet.VisualStudio.Common/SupportedProjectTypes.cs

```diff
@@ -74,5 +83,10 @@ public static bool IsSupportedForAddingReferences(string projectKind)
         {
             return !_unsupportedProjectTypesForAddingReferences.Contains(projectKind, StringComparer.OrdinalIgnoreCase);
         }
+
+        public static bool IsSupportedProjectExtension(string projectPath)
+        {
+            return !_unsupportedProjectExtension.Contains(Path.GetExtension(projectPath));
```

Comment:
Path.GetExtension can return null if the path does not have an extension, does hashset handle that?

Reply:
Yes, HashSet return false if you pass null